### ISSUE TITLE
Issue on new language fine-tuning

### ISSUE THREAD

**Issue by juliee235** [OpenBMB/VoxCPM#195]
I fully fine-tuned the model on a new language using a combined dataset of around 9,000 hours of Thai and English. The results are generally good, but there’s an issue where the model sometimes fails to stop generation. I suspect this is due to a conflict between the two loss functions, where the stop loss converges much faster than the FM loss. Any suggestions?

<img width="970" height="322" alt="Image" src="https://github.com/user-attachments/assets/e4b97982-fd46-4855-be81-3de314ed357d" />

The light blue curve corresponds to λ_diff = 1.0 and λ_stop = 1.0, while the purple curve corresponds to λ_diff = 1.0 and λ_stop = 0.1.

general setting:
batch size : 4
grad_accum_steps: 8
num_gpu : 4
lr : 1e-5
weight_decay: 0.01
warmup_steps: 10000
num_iters: 300000

**Comment by liuxin99** [comment]
Hi, it does look a bit strange. Maybe you could try increasing the weight of the stop loss and see how it affects the results. In addition, it might help to check whether there are any abnormal samples in your data, for example audio clips with very long silence at the end.

**Comment by juliee235** [comment]
Thanks for your response. Just to clarify—do you mean increasing lambda_stop to be higher than lambda_diff?

**Comment by liuxin99** [comment]
Yeah, I think you could try increasing the weight of the stop loss based on your already fine‑tuned model and see whether this helps bring the stop loss down. In addition, it is still recommended to carefully inspect the dataset to make sure there are no potential issues in the data itself. Moreover, if this phenomenon does not occur very frequently, you might also consider mitigating it by referring to the retry mechanism here: https://github.com/OpenBMB/VoxCPM/blob/main/src/voxcpm/model/voxcpm.py#L657-L693.

**Comment by juliee235** [comment]
I checked my dataset and found that only about 5,000 out of 2,500,000 samples have more than 0.5 seconds of silence at the end, so I don’t think that’s the cause of the issue. Could you share how you set lambda_stop and lambda_diff during training?

**Comment by liuxin99** [comment]
In our pre-training setting, the loss weights are also lambda_stop = 1 and lambda_diff = 1, and we rarely encounter the issue where the model fails to stop generation.
